src/ngraph/runtime/inference_engine/ie_executable.cpp

```diff
@@ -0,0 +1,143 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/inference_engine/ie_executable.hpp"
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/pass/manager.hpp"
+#include "ngraph/pass/opset1_upgrade.hpp"
+#include "ngraph/runtime/inference_engine/ie_tensor_view.hpp"
+
+ngraph::runtime::inference_engine::IE_Executable::IE_Executable(std::shared_ptr<Function> func,
+                                                                std::string _device)
+{
+    auto opset = ngraph::get_opset1();
+    pass::Manager passes;
+    passes.register_pass<pass::Opset1Upgrade>();
+    passes.run_passes(func);
+
+    for (auto& node : func->get_ops())
+    {
+        if (!opset.contains_op_type(node.get()))
+        {
+            if (node->get_type_info() == op::GetOutputElement::type_info)
+            {
+                // IE currently can handle GetOutuputElement op;
+                continue;
+            }
+            else
+            {
+                std::cout << "UNSUPPORTED OP DETECTED: " << node->get_type_info().name << std::endl;
+                THROW_IE_EXCEPTION << "Detected op not belonging to opset1!";
+            }
+        }
+    }
+
+#ifdef NGRAPH_DEBUG_ENABLE
+    std::cout << "Nodes in test: ";
+    for (auto& node : func->get_ops())
+    {
+        std::cout << node << std::endl;
+    }
+    std::cout << std::endl;
+#endif
+
+    network = InferenceEngine::CNNNetwork(func);
+    device = _device == "INFERENCE_ENGINE" ? "CPU" : _device;
+    set_parameters_and_results(*func);
+}
+
+bool ngraph::runtime::inference_engine::IE_Executable::call(
+    const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+    const std::vector<std::shared_ptr<runtime::Tensor>>& inputs)
+{
+    InferenceEngine::Core ie;
+
+    //  Loading model to the plugin (BACKEND_NAME)
+    InferenceEngine::ExecutableNetwork exeNetwork = ie.LoadNetwork(network, device);
+    //  Create infer request
+    InferenceEngine::InferRequest inferRequest = exeNetwork.CreateInferRequest();
+    //  Prepare input and output blobs
+    InferenceEngine::InputsDataMap inputInfo = network.getInputsInfo();
+
+    if (inputInfo.size() != inputs.size())
+    {
+        THROW_IE_EXCEPTION << "Function inputs number differ from number of given inputs";
+    }
+
+    size_t i = 0;
+    for (auto& it : inputInfo)
+    {
+        std::shared_ptr<runtime::inference_engine::IETensorView> tv =
+            std::static_pointer_cast<runtime::inference_engine::IETensorView>(inputs[i]);
+        size_t size = tv->data.size() / sizeof(float);
+        float* orig_data = (float*)tv->data.data();
+        std::vector<float> data(orig_data, orig_data + size);
```

Comment:
This data copy is not needed. You can change fill_blob to take float* and size and copy that way. Copies are very expensive.